### PR TITLE
Update required dependencies for Debian/Ubuntu

### PR DESCRIPTION
**Pull request by mfonville**
After trying to build Atom on a very bare Ubuntu install, I figured out some missing dependencies for building: https://github.com/atom/flight-manual.atom.io/pull/683

That also implied these libraries are the actual correct dependencies

### CODE REVIEW

**Comment by mfonville** [comment]
@DeeDeeG would you maybe mind checking these for the Red Hat version?

**Comment by DeeDeeG** [comment]
@mfonville I know it's likely too late to ask, but it would be great if you could provide any error messages that let you know that these packages were needed. Or the names/paths of the needed files, for instance shared object (`.so`) files. (As I recall we found out in the previous PR, the package names may not be the same in `.rpm` land, but the shared object files have a fair chance of having the same name.)

Or not quite as time-saving, but what minimal environment did you use to find out these packages were needed? (The `debian` or `ubuntu` Docker images, maybe?)

**Review by Codingprodigy-tech — APPROVED** [review]

**Comment by mfonville** [comment]
@DeeDeeG I unfortunately don't have the exact .so anymore. You can take a look at the PR for the Atom manual I submitted, in addition to that there were only some libraries that I did find, but later seemed to be implied by e.g `gtk3` like: `libxcomposite1` `libxcursor1` `libxdamage1` `libatk1.0-0` `libatk-bridge2.0-0` `libgdk-pixbuf2.0-0`; and `libxtst6` did imply already for `libxi6` 

The testing I did by building Atom on a Ubuntu Server ISO (20.04 in Virtualbox)

**Comment by mfonville** [comment]
@DeeDeeG could you check if #22076 works for you on RPM-based distros? Then both PRs can be merged.

**Comment by DeeDeeG** [comment]
I haven't really had the time to set aside and take a look at this. I'm afraid I might not for a couple of weeks.

**Comment by mfonville** [comment]
Ok, there is no hurry :-)

**Comment by Imreallycool8** [comment]
Is my inbox some sort of message system?

**Comment by DeeDeeG** [comment]
@Imreallycool8 perhaps you are subscribed to updates in this repository?

Look for this button at the top of the page:

GitHub "watch" button

You can click that and select a narrower option, such as "Participating and @mentions" or "Ignore" or "Custom".

**Comment by Imreallycool8** [comment]
Hello, how do I make a new window for just a text box?

**Comment by mfonville** [comment]
@darangi You can already safely merge this PR, the other PR for the RPM-dependencies does not have any direct dependency with this one.

Thank you

**Review by sadick254 — APPROVED** [review]

File: resources/linux/debian/control.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: <%= appFileName %>
 Version: <%= version %>
-Depends: git, libgtk-3-0 (>= 3.9.10), libgcrypt20, libnotify4, libxtst6, libnss3 (>= 2:3.22), libglib2.0-bin | kde-cli-tools | kde-runtime, xdg-utils, libx11-xcb1, libxss1, libxkbfile1, libcurl3 | libcurl4
+Depends: git, libgcrypt20, libgtk-3-0 (>= 3.9.10), libnotify4, libnss3 (>= 2:3.22), libglib2.0-bin | kde-cli-tools | kde-runtime, xdg-utils, libx11-xcb1, libxcb-dri3-0, libxss1, libxtst6, libxkbfile1, libcurl3 | libcurl4
```
Comment (DeeDeeG) [inline resources/linux/debian/control.in:3]: Do we actually need `libX11-dri-3-0` or is it `libX11-xcb` like on Fedora/openSUSE?

Maybe `libX11-xcb1`? See: https://packages.ubuntu.com/hirsute/amd64/libx11-xcb1/filelist

Comment (DeeDeeG) [reply resources/linux/debian/control.in:3]: Okay, I see `libxcb-dri3-0` is actually needed, from some testing in Docker.

**Comment by DeeDeeG** [comment]
I'm giving this the old "Docker test". Results later today, if all goes to plan.

**Comment by DeeDeeG** [comment]
I can recommend adding `libgbm1` to the requirements. Once that is installed, all tested Debian/Ubuntu versions can do `atom-dev --version --no-sandbox`.

Results:

- Debian
  - 8-slim (End-of-Life): needs `libgbm.so.1` --> `libgbm1` package :x:
  - 9-slim (oldest supported): good, no changes needed :heavy_check_mark:
  - 10-slim (newest supported): needs `libgbm.so.1` --> `libgbm1` package :x:
- Ubuntu
  - 16.04 (recently End-of-Life): good, no changes needed :heavy_check_mark:
  - 18.04 (oldest supported): needs `libgbm.so.1` --> `libgbm1` package :x:
  - 20.04: good, no changes needed :heavy_check_mark:
  - 20.10: good, no changes needed :heavy_check_mark:
  - 21.04 (newest supported): good, no changes needed :heavy_check_mark:

<details><summary>Testing method and notes (click to expand):</summary>

Similar to https://github.com/atom/atom/pull/22076#issuecomment-843338497...

Once the Docker container is up and running, first do `apt-get update && apt-get install nano` to get a working text editor. Use `nano` to write out and save this script, or else `apt-get` will swallow some of the later commands and they won't actually run:

```sh
export DEBIAN_FRONTEND=noninteractive # skip manual configuration prompt during install of the `tzdata` package
apt-get install -y wget unzip
wget 'https://dev.azure.com/DeeDeeG/b/_apis/build/builds/1128/artifacts?artifactName=atom-amd64.deb&api-version=6.0&%24format=zip' -O _atom.zip
unzip _atom.zip
apt-get install -y ./atom-amd64.deb/atom-amd64.deb
atom-dev --version --no-sandbox
/usr/share/atom-dev/atom --version --no-sandbox
apm-dev --version
```

Then run this script with `bash ./[script_filename_here]`.

Note: For the now End-of-Life Debian 8 release, `apt-get` refuses to install a bare `.deb` package. So you have to use `dpkg -i ./atom-amd64.deb/atom-amd64.deb` then `apt-get install -f` to get the downloaded Atom `.deb` file installed, instead of `apt-get install ./atom-amd64.deb/atom-amd64.deb`. Then the rest of the script is the same.

</details>

**Comment by mfonville** [comment]
Ok, I added `libgbm1` too.

Also I changed `libasound2` from recommends to requires, because it is after further checking a strict requirement. Atom cannot start without it.

**Review by DeeDeeG — APPROVED** [review]
Looks good to me. :+1: 

I re-tested Debian 10 and Ubuntu 18.04, which had a missing dependency before, and I also re-tested Ubuntu 20.04 for good measure. All pass the "Docker test" now (test described above). I'm not re-testing the others as they all should pass now, and it takes too long to test all eight.

**Comment by sadick254** [comment]
@DeeDeeG Thank you for the additional tests.

**Review by NOUIY — APPROVED** [review]

**Comment by CyberzNet** [comment]
> After trying to build Atom on a very bare Ubuntu install, I figured out some missing dependencies for building: [atom/flight-manual.atom.io#683](https://github.com/atom/flight-manual.atom.io/pull/683)
> 
> That also implied these libraries are the actual correct dependencies

https://github.com/atom/atom/pull/22015#issue-825034793

**Pull request merged**